packages/vite/src/node/logger.ts

```diff
@@ -165,19 +166,31 @@ export function createLogger(
   return logger
 }
 
-export function printServerUrls(
+export function printServerInfo(
   urls: ResolvedServerUrls,
   optionsHost: string | boolean | undefined,
+  mode: string,
+  envDir: string,
   info: Logger['info'],
 ): void {
-  const colorUrl = (url: string) =>
-    colors.cyan(url.replace(/:(\d+)\//, (_, port) => `:${colors.bold(port)}/`))
+  const formatUrl = (url: string) =>
+    url.replace(/:(\d+)\//, (_, port) => `:${colors.bold(port)}/`)
+
   for (const url of urls.local) {
-    info(`  ${colors.green('➜')}  ${colors.bold('Local')}:   ${colorUrl(url)}`)
+    info(
+      `  ${colors.green('➜')}  ${colors.bold('Local')}:   ${colors.cyan(formatUrl(url))}`,
+    )
   }
   for (const url of urls.network) {
-    info(`  ${colors.green('➜')}  ${colors.bold('Network')}: ${colorUrl(url)}`)
+    info(
+      `  ${colors.green('➜')}  ${colors.bold('Network')}: ${colors.cyan(formatUrl(url))}`,
+    )
   }
+  info(`  ${colors.green('➜')}  ${colors.bold('Mode')}:    ${mode}`)
+  const envFiles = getLoadedEnvFileNamesForMode(mode, envDir)
+  info(
+    `  ${colors.green('➜')}  ${colors.bold('Env')}:     ${envFiles.length ? envFiles.join(' ') : 'no env files loaded'}`,
+  )
```

Comment:
@sapphi-red says 

> I think it makes sense to show the mode. But for the loaded env files, I guess you can know which file is loaded if you know which mode it's running in.

https://github.com/vitejs/vite/issues/18719#issuecomment-2489995231

---

Personally, it's user-friendly to show which `.env` files are loaded in what order, especially for beginners and teams where both senior and junior are, even they are described in documentation.

Adding this feature doesn't seem to make Vite such slow, does it?

Reply:
I don't think it'll affect the perf. I felt showing the env files is a bit too wordy. Although, that's probably just because I'm too familiar with Vite 😅 I think I'm fine if the files are listed in a single line. Even further, maybe it can be in the same line with Mode?

Reply:
I personally would not print the env files. If someone is interested in knowing that (usually for debugging purposes), we could probably print it under `vite:config` or `vite:env` DEBUG flags if they're not already printed.